girder/girder_large_image/rest/large_image_resource.py

```diff
@@ -445,6 +447,55 @@ def _deleteCachedImages(self, spec, associatedImages=False, imageKey=None):
                 removed += 1
         return removed
 
+    @access.user(scope=TokenScope.DATA_WRITE)
+    @autoDescribeRoute(
+        Description('Create large images for all items within a folder.')
+        .notes('Does not work for items with multiple files.')
+        .modelParam('id', 'The ID of the folder.', model=Folder, level=AccessType.WRITE, required=True)
+        .param('createJobs','Whether a job should be created for each large image.', required=False, 
+            default=False, dataType='boolean')
+        .param('localJobs', 'Whether a the jobs created should be local.', required=False, 
+            default=False, dataType='boolean')
+        .param('recurse', 'Whether child folders should be recursed', required=False, default=False, 
+            dataType='boolean')
+        .errorResponse('ID was invalid.')
+        .errorResponse('Write access was denied for the folder.', 403)
+    )
+    def createLargeImages(self, folder, params):
+        user=self.getCurrentUser()
+        createJobs = params.get('createJobs')
+        if createJobs:
+            createJobs = 'always' or True
+        self.createImagesRecurseOption(folder=folder, createJobs=createJobs, user=user, 
+                                recurse=params.get('recurse'), localJobs=params.get('localJobs'))
+                    
+    def createImagesRecurseOption(self, folder, createJobs, user, recurse, localJobs):
+        count = {'large images created': 0}
+        if recurse:
+            for childFolder in Folder().childFolders(parent=folder, parentType='folder'):
+                self.createImagesRecurseOption(folder=childFolder, createJobs=createJobs, user=user, 
+                                        recurse=recurse, localJobs=localJobs)
+        for item in Folder().childItems(folder=folder):
+            if item.get('largeImage'):
+                if item['largeImage'].get('expected'):
+                    pass
+                else:
+                    try:
+                        Item().getMetadata(item)
+                        continue 
+                    except Exception:
+                        previousFileId = item['largeImage'].get('originalId', item['largeImage']['fileId'])
+                        ImageItem().delete(item)
+                        ImageItem().createImageItem(item, File().load(user=user, id=previousFileId), 
+                                                    createJob=createJobs, localJob=localJobs)
+            else:
+                largeImageFileId = None
+                files = list(Item().childFiles(item=item, limit=0))
```

Comment:
Did you ever find out if we can document the behavior of `limit=0` somewhere?

Reply:
It could probably be added in the method's docstring, but it's defined in Girder.

Reply:
Ok, thanks. It's unrelated to this PR, so I'll mark as resolved